GAE/src/org/akvo/flow/servlet/ReportServlet.java

```diff
@@ -0,0 +1,196 @@
+            case ReportTaskRequest.START_ACTION:
+                Report r = rDao.getByKey(id);
+                if (r != null) {
+                    if (!r.getState().equals(Report.QUEUED)) {
+                        //wrong state
+                        log.warning("Cannot start report that is " + r.getState());
+                        //TODO do anything else?
+                        return null;
+                    }
+                    //look up user
+                    final String email = uDao.getByKey(r.getUser()).getEmailAddress();
+
+                    //Gleaned from export-reports-views.js
+                    ReportBody body = new ReportBody();
+                    body.criteria = new ReportCriteria();
+                    body.criteria.opts = new ReportOptions();
+                    body.criteria.appId = PropertyUtil.getProperty("appId");
+                    body.criteria.email = email;
+                    body.criteria.surveyId = r.getFormId();
+                    body.criteria.appId = PropertyUtil.getProperty("appId");
+                    body.criteria.exportType = r.getReportType();
+                    body.criteria.opts.exportMode = r.getReportType();
+                    body.criteria.opts.reportId = r.getKey().getId();
+                    body.criteria.opts.from = r.getStartDate();
+                    body.criteria.opts.to = r.getEndDate();
+                    body.criteria.opts.lastCollection = r.getLastCollectionOnly();
+                    body.criteria.opts.questionId = r.getQuestionId();
+                    body.criteria.opts.imgPrefix = PropertyUtil.getProperty("photo_url_root");
+                    body.criteria.opts.uploadUrl = PropertyUtil.getProperty("surveyuploadurl");
+                    ObjectMapper objectMapper = new ObjectMapper();
+                    byte[] postData = objectMapper.writeValueAsBytes(body);//UTF-8?
+
+
+                    //hit the services server
+                    try {
+                        URL url = new URL(PropertyUtil.getProperty("flowServices") + "/generate");
+                        HttpURLConnection con = (HttpURLConnection) url.openConnection();
+                        con.setRequestMethod("POST");
+                        con.setRequestProperty("Content-Type", "application/json");
+                        con.setRequestProperty("charset", "utf-8");
+                        log.info("Preparing to POST " + body.toString() + " to  " + url);
+                        try( DataOutputStream wr = new DataOutputStream( con.getOutputStream())) {
+                           wr.write( postData );
+                        }
+                        final int sts = con.getResponseCode();
+                        log.info(" got  " + sts);
+
+                        if (sts == 200) {
+                            //Success, we are done!
+                            return null;
+                        }
+                        //if permanent error, fail this report
```

Comment:
The formatting feels a bit weird here. maybe this comment should go inside the `else if` branch?

Reply:
yes